src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -36,14 +37,49 @@ internal IArgument CreateArgumentOperation(ArgumentKind kind, IParameterSymbol p
         {
             var value = Create(expression);
 
+            // put argument syntax to argument operation
+            var argumentExist = value.Syntax?.Parent is ArgumentSyntax;
+
+            // if argument syntax doesn't exist, this operation is implicit
             return new CSharpArgument(kind,
                 parameter,
                 value,
                 semanticModel: _semanticModel,
-                syntax: value.Syntax,
+                syntax: argumentExist ? value.Syntax.Parent : value.Syntax,
                 type: value.Type,
                 constantValue: default,
-                isImplicit: expression.WasCompilerGenerated);
+                isImplicit: expression.WasCompilerGenerated || !argumentExist);
+        }
+
+        private IVariableDeclaration CreateVariableDeclaration(BoundLocalDeclaration boundLocalDeclaration, SyntaxNode syntax)
+        {
+            return OperationFactory.CreateVariableDeclaration(boundLocalDeclaration.LocalSymbol, Create(boundLocalDeclaration.InitializerOpt), _semanticModel, syntax);
+        }
+
+        private IOperation CreateBoundCallInstanceOperation(BoundCall boundCall)
+        {
+            if (boundCall.Method == null || boundCall.Method.IsStatic)
+            {
+                return null;
+            }
+
+            return Create(boundCall.ReceiverOpt);
+        }
+
+        private IEventReferenceExpression CreateBoundEventAccessOperation(BoundEventAssignmentOperator boundEventAssignmentOperator)
+        {
+            SyntaxNode syntax = boundEventAssignmentOperator.Syntax;
+            // BoundEventAssignmentOperator doesn't hold on to BoundEventAccess provided during binding.
+            // Based on the implementation of those two bound node types, the following data can be retrieved w/o changing BoundEventAssignmentOperator:
+            //  1. the type of BoundEventAccess is the type of the event symbol.
+            //  2. the constant value of BoundEventAccess is always null.
+            //  3. the syntax of the boundEventAssignmentOperator is always AssignmentExpressionSyntax, so the syntax for the event reference would be the LHS of the assignment.
+            IEventSymbol @event = boundEventAssignmentOperator.Event;
+            Lazy<IOperation> instance = new Lazy<IOperation>(() => Create(boundEventAssignmentOperator.Event.IsStatic ? null : boundEventAssignmentOperator.ReceiverOpt));
```

Comment:
Is the `IsStatic` check necessary?

Reply:
I just moved this code from CSharpOperationFactory to this file. 

tagging @genlu who wrote the code.

Reply:
opened https://github.com/dotnet/roslyn/issues/22233